src/main/java/reposense/model/AuthorConfiguration.java

```diff
@@ -145,10 +145,27 @@ public void addAuthor(Author author, List<String> ignoreGlobList) {
     public void removeIgnoredAuthors(List<String> ignoredAuthorsList) {
         for (String author : ignoredAuthorsList) {
             if (authorEmailsAndAliasesMap.containsKey(author)) {
-                authorList.remove(authorEmailsAndAliasesMap.get(author));
+                removeAuthor(author);
             }
         }
-        resetAuthorInformation();
+    }
+
+    /**
+     * Removes all information of the {@code author} from the configs
+     * @param author Can be an author's git ID, email, or alias
+     */
+    public void removeAuthor(String author) {
+        Author ignoredAuthor = authorEmailsAndAliasesMap.get(author);
+        authorList.remove(ignoredAuthor);
+        authorDisplayNameMap.remove(ignoredAuthor);
+        authorEmailsAndAliasesMap.remove(author);
+        authorEmailsAndAliasesMap.remove(ignoredAuthor.getGitId());
```

Comment:
A clarification: if `author` is the same as `gitID`, then we'll be removing the key-value binding twice.
Would this be an issue?
Is it simpler to omit line 161 entirely, since we're also removing the key-value bindings for git ID, aliases and emails?

Reply:
We can do this, I think. Just that the current one makes sure, nothing is left, at all. Let me take a check on the code and get back on this.

Reply:
**Update:** Have removed it. We are removing the rest explicitly. So should be fine.